# Aufgaben
1. Ermitteln Sie die Gitterkonstante eines Reflexionsgitters.
Dazu sind das reflektierte Licht (0. Ordnung) sowie die ersten drei Beugungsordnungen der grünen
Linie ($\lambda$ = 546.1 nm) einer Hg-Spektrallampe bei einer festen Gitteranordnung (etwa 60° zwischen
einfallendem Licht und Gitternormale) auszumessen.
2. Bestimmen Sie die acht Wellenlängen von vier Dubletts des Spektrums einer Na-Spektrallampe aus
dem Beugungsspektrum 1. Ordnung. Es ist dieselbe Gitteranordnung wie in Aufgabe 1 zu wählen.
3. Berechnen Sie das Auflösungsvermögen des Reflexionsgitters.
Die wirksame Gitterbreite beträgt 50 mm (0.5%).

# Aufgabe 1
## Experimentaufbau
![Skizze vom Versuchsaufbau0](./Skizze.jpg){#fig-skizze}

In unserem Experiment hatten wir, mithilfe des Präzisonsgoniometer-Spektrometer, $\varphi$ gemessen. $\alpha$ ist wie Abbildung 1 zeigt der Einfalswinkel ist, während $\beta$ der Ausfallswinkel ist. Dabei ist $\alpha$ über das ganze Experiment über konstant und nur $\varphi$ und $\beta$ ändern ihre größe. 

Um $\beta$ zu erhalten muss man $\beta = 180° - \varphi - \varphi_0 - \alpha$ berechnen. Der Winkel $\alpha$ ergibt sich dabei, wenn der Winkel $\varphi$ auf $\approx 60°$ eingestellt wird und das Maximum 0ter Ordnung mit dem Spiegel eingestellt wird. Dann gilt nähmlich $\alpha = \beta$, da der gesammt Strahl reflektiert wird, so kommt man auf die Formel, wobei $\varphi_{max} = 60.09°$ dem Winkel der ersten Ordnung entspricht. 
$$\alpha = \frac{180° - \varphi_{max} - \varphi_0}{2} = 60.1581°$$

Da das Messgerät nicht genau auf der Nullstelle war als der Strahl das Fadenkreuz geschnitten hat, wurde ein Korrekturswinkel gemessen um die später aufgenommenen Daten zu korrigieren. Dieser lag bei uns, bei $359.5939°$ um nun den Korrekturwinkel $\varphi_0$ zu erhalten, rechnen wir folgendermaßen.
$$\varphi_0 = 360° - 359.5939° = -0.4061°$$
Da dies jedoch schwer einzuzeichnen war, ist dieser Winkel nicht in der Skizze zu finden.

## Ermittelung der Gitterkonstante
In der ersten Aufgabe ging es darum die Gitterkonstante des Reflexionsgitters zu ermitteln. Dabei nutzeten wir eine die grünen Spektrallinien, mit einer Wellenlänge von $\lambda = 546.1$ nm einer Hg-Spektrallampe. Ermittelt wurde $\varphi$ sowie der Korrekturwinkel.

Um nun auf die die Gitterkonstante zu kommen wird, die aus der Beschreibung des Experimentes die folgende Formel, der Phasendifferenz genommen.
$$\delta = 2 \pi \frac{g(sin \alpha - sin \beta)}{\lambda}$$
Dabei kann $\delta$ geschrieben werden als $\delta = 2k\pi$, wobei $k$ die Beugungsordnung ist ^["Diese Formel wurde aus dem Buch "Physikalischen Praktikum", 14. Auflage, Hrsg. W. Schenk, F. Kremer, Optik, 2.3, S.263 entnommen"]. So lässt sich die vorherige Formel umstellen nach $g$.
$$g = \frac{k \lambda}{sin \alpha - sin \beta}$$
Mit dieser Formel, kann man nun die Gitterkonstante für jede Messung ausrechnen, was in der folgenden Tabelle gemacht wurde.

In der Tabelle wurde mit den Rohwerten und mit den Korrekturwinkel $\varphi_0$ die Gitterkonstante $g$ berechnet.


| Beugungsordn. | $\alpha$ | $\beta$ | $\varphi$ | $\varphi_0$ | g ($10^{-6} m$) |
|------------|------------|------------|------------|------------|------------|
| 0te | $59.955° \pm 2''$ | $59.955°  \pm 2'' $ | $60.09°\pm 2''$ | $0°$ | - |
| 0te mit Korrekturwinkel | $60.158° \pm 2''$ | $60.158° \pm 2''$ | $59.684° \pm 2''$ | $0.4061° \pm 2''$ | - |
|------------|------------|------------|------------|------------|------------|
| 1te | $59.955° \pm 2''$ | $30.775° \pm 2''$ | $89.27°\pm 2''$ | $0°$ | $1.543$ |
| 1te mit Korekturwinkel | $60.158° \pm 2''$ | $30.978° \pm 2''$ | $88.864° \pm 2''$ | $0.4061° \pm 2''$ | $1.548$ |
|------------|------------|------------|------------|------------|------------|
| 2te | $59.955° \pm 2''$ | $8.985° \pm 2''$ | $111.06°\pm 2''$ | $0°$ | $1.539$ |
| 2te mit Korekturwinkel | $60.158° \pm 2''$ | $9.188° \pm 2''$ | $110.654° \pm 2''$ | $0.4061° \pm 2''$ | $1.543$ |
|------------|------------|------------|------------|------------|------------|
| 3te | $59.955° \pm 2''$ |  $-11.475° \pm 2''$ | $131.52°\pm 2''$ | $0°$ | $1.539$ |
| 3te mit Korekturwinkel |  $60.158° \pm 2''$ | $-11.272° \pm 2''$ | $131.114° \pm 2''$ | $0.4061° \pm 2''$ | $1.541$ |

: Tabelle mit allen bestimmten Winkeln, von der 0ten und 3ten Ordnung und die jeweilige Ermittelte Gitterkonstante

Es ergibt sich so aus der Tabelle eine durchschnittliche Gitterkonstante von $1.540 \cdot 10 ^{-6}m$ ohne Korrekturwinkel und eine durchschnittliche Gitterkonstante von $1.544 \cdot 10 ^{-6}$. Dabei gilt ein relativer Fehler mit der Fehlerfortpflanzung von 0.15%.


In [1]:
# Gitterkonstante für alle Winkel
import numpy as np

korrektur_winkel = 0.4061

winkel_pro_ordnungszahl = np.array([89.27, 111.06, 131.52]) #phi
winkel_pro_ordnungszahl = winkel_pro_ordnungszahl - korrektur_winkel
erste_ordnung = 60.09 - korrektur_winkel #phi_0

alpha = (180 - erste_ordnung)/2 #minus korre
_lambda = 546.1 * 10**(-9)
def g(winkel, k):
    return (k * _lambda)/(np.sin(alpha * np.pi/180)-np.sin(winkel * np.pi/180))

gitter_konst = np.array([])

for i in range(len(winkel_pro_ordnungszahl)):
    print("BETA: ", 180 - winkel_pro_ordnungszahl[i] - alpha)
    gitter_konst = np.append(gitter_konst, g(180 - winkel_pro_ordnungszahl[i] - alpha, i + 1))

#print(gitter_konst)
print("Gemittelte Gitterkonstnate:")
print(np.mean(np.abs(gitter_konst)))
print("Berechnetes Alpha:")
print(alpha)

#print(winkel_pro_ordnungszahl)

BETA:  30.978049999999996
BETA:  9.18804999999999
BETA:  -11.271950000000004
Gemittelte Gitterkonstnate:
1.5443422436719798e-06
Berechnetes Alpha:
60.15805


## Fehlerbetrachtung und Diskussion
Da das Ablesen der Daten, optisch erfolgte, kann es zu Paralaxefehlern gekommen sein, da es nicht möglich ist, durch das Beobachtungsrohr perfekt zu schauen, kann es möglich sein, dass die Spektrallinie nicht immer genau in der Mitte des Fadenkreuz gewesen ist. Das gleiche gilt für das ablesen ders Gradmaßes, wobei dieses auch einen eigenen Messgenauigkeitsfehler von 2 Sekunden hat. Zudem war die Skala zum auslesen, vorallem die für die Minuten und Sekunden, an einigen stellen etwas verschmutzt was das Ablesen erschwierigte.

Der maximale Fehler aus der Tabelle wurde mit einer Fehlerfortpflanzung berechnet, wobei für die Winkel eine Messunsicherheit von $2''$ Sekunden genommen wurde.

Wie in der Tabelle zu sehen ist, kommen wir auf einen Durchschnittlichen Wert für die Gitterkonstante auf $g = 1.540 \cdot 10^{-6}m$ bzw. mit Korrekturwinkel von $g = 1.544 \cdot 10^{-6}m$ welcher sehr plausiebel ist. Da andere Reflexionsgitter, ähnliche Gitterkonstanten vorweisen.


# Aufgabe 2

Für die zweite Aufgabe wurde die Hg-Spektrallampe ausgetauscht für eine Na-Spektrallampe, außerhalb davon blieb der Aufbau gleich. Es wurde wieder damit begonnen, den Korrekturwinkel $\phi_0$ zu bestimmen. Für diesen haben wir $\phi_0 = 359,5928°$ gemessen, oder auch $\phi_0 = -0,4072°$. Genau wie in Aufgabe 1 wurde auch das Reflexionsgitter so eingestellt, das bei ca. 60° das Maximum 0. Ordnung zu sehen ist. Genauer liegt dies bei $Max_0 = 60,1078°$.

Bei der Na-Spektrallampe kann man nicht nur einzelen Maxima sehen sondern durch die Spin-Bahn-Kopplung bei Natrium kann man 4 verschiedene Dubletts erkennen in den Farben Rot, Orange, Grün und Blau. Die Maxima erster Ordnung von allen Dubletts wurden gemessen und folgende Werte haben wir abgelesen:

| Ordnung  | Grad [°] | Minuten ['] | Sekunden [''] | Gesamt [°]    |
|----------|---------:|------------:|---------------:|------------:|
| Blau 1   |       87 |          16 |            28 $\pm 2$  | 87,27444444 $\pm 2''$|
| Blau 2   |       87 |          17 |            32 $\pm 2$  | 87,29222222 $\pm 2''$|
| Grün 1   |       90 |          21 |            22  $\pm 2$ | 90,35611111 $\pm 2''$|
| Grün 2   |       90 |          22 |            36  $\pm 2$ | 90,37666667 $\pm 2''$|
| Orange 1 |       91 |          14 |            18  $\pm 2$ | 91,23833333 $\pm 2''$|
| Orange 2 |       91 |          15 |            28  $\pm 2$ | 91,25777778 $\pm 2''$|
| Rot 1    |       92 |          21 |            28  $\pm 2$ | 92,35777778 $\pm 2''$|
| Rot 2    |       92 |          22 |            48  $\pm 2$ | 92,38000000 $\pm 2''$|

: Ermittelter Winkel in Grad, Minuten und Sekunden der verschieden Farbigen Dobletts

Da bei $Max_0$ gilt $\alpha = \beta$, können wir wieder mit der Formel $\alpha = \frac{1}{2}(180° - (Max_0 - \phi_0))$ berechnen. Daraus folgt $\alpha = \frac{1}{2}(180° - (60,1078° + 0,4072))= 59,7425°$

Für die Berechnung der Wellenlänge nutzen wir die Formel: 
$$m \cdot \lambda = g \cdot (sin(\alpha) - sin(\beta))$$
wobei wir immer nur das erste Maximum betrachten, weshalb immer m=1 gilt. Den Winkel für $\beta$ können wir berechnen mit $\beta = 180° - (\alpha + (\phi - \phi_0))$.

Für die Gitterkonstante g nutzen wir den aus Aufgabe 1 berechneten Wert: $g = 1.544 \cdot 10 ^{-6}m$

In [2]:
import math
import numpy as np

# Winkelwerte in Grad
blau_1 = 87.27444444
blau_2 = 87.29222222
gruen_1 = 90.35611111
gruen_2 = 90.37666667
orange_1 = 91.23833333
orange_2 = 91.25777778
rot_1 = 92.35777778
rot_2 = 92.38

korrekturwinkel = 0.4072
alpha = 59.7425
d = 1.544e-6  # Gitterkonstante in Metern

def get_wavelength(phi):
    beta = 180 - (alpha + (phi - korrekturwinkel))  # Richtige Formel!
    wavelength = d * (np.sin(alpha * np.pi/180) - (np.sin(beta * np.pi/180))) * 1e9
    return wavelength

def main():
    colors = {
        "blau_1": blau_1,
        "blau_2": blau_2,
        "gruen_1": gruen_1,
        "gruen_2": gruen_2,
        "orange_1": orange_1,
        "orange_2": orange_2,
        "rot_1": rot_1,
        "rot_2": rot_2
    }

    for name, angle in colors.items():
        wl = get_wavelength(angle)
        print(f"{name}: {wl:.3f} nm")

if __name__ == "__main__":
    main()


blau_1: 483.937 nm
blau_2: 484.337 nm
gruen_1: 554.470 nm
gruen_2: 554.948 nm
orange_1: 575.086 nm
orange_2: 575.542 nm
rot_1: 601.504 nm
rot_2: 602.031 nm


Somit haben wir für unsere verschiedenen Dubletts die folgenden berechneten Wellenlängen:

| Ordnung  | Gesamt [°]  | Wellenlänge [nm]
|----------|------------:|------------:|
| Blau 1   | 87,27444444 | 483.937 |
| Blau 2   | 87,29222222 | 484.337 |
| Grün 1   | 90,35611111 | 554.470 |
| Grün 2   | 90,37666667 | 554.948 |
| Orange 1 | 91,23833333 | 575.086 |
| Orange 2 | 91,25777778 | 575.542 |
| Rot 1    | 92,35777778 | 601.504 |
| Rot 2    | 92,38000000 | 602.031 |

: Berechnete Wellenlnge, zu bestimmten Winkel

## Vergleich mit Literaturwerten und Diskussion


Wir vergleichen nun unsere Werte mit den Literaturwerten für das Natrium Spektrum, dafür nutzen wir die Tabelle aus "W. Schenk, F.Kremer, Physikalisches Praktikum, 14. Auflage (Springer, Heidelberg, 2014), S. 385". In der Tabelle unten hinzugefügt sind auch die Abweichnungen zu unseren berechneten Wellenlängen:


| Farbe| Wellenlänge 1 [nm] | Wellenlänge 2 [nm] | Abweichung 1 | Abweichung 2 |
| :---------------- | -----------------: | -----------------: | -----------: | -----------: |
| **blau**          |              514,4 |              514,9 |       5,92 % |       5,94 % |
| **gelbgrün**      |              568,3 |              568,7 |       2,43 % |       2,42 % |
| **orange**        |             589,00 |             589,59 |       2,36 % |       2,38 % |
| **rot**           |              615,3 |              616,0 |       2,24 % |       2,27 % |

: Berechnete Wellenlängen mit zugehörigen Abweichungne zum Literaturwert.


Die gemessenen Wellenlängen der Natrium-Dubletts liegen im sichtbaren Bereich und zeigen den erwarteten Trend steigender Wellenlänge mit zunehmender Ordnung. Im Vergleich zu den Literaturwerten aus der Literaturtabelle betragen die Abweichungen bis auf blau unter 3%. Die Unterschiede lassen sich auf Justageungenauigkeiten, Messfehler bei den Winkeln sowie eine leicht abweichende Gitterkonstante zurückführen. Beim blauen Dublett ist die Abweichung mit 6% wesentlich höher als bei den anderen, was daher kommen könnte, dass die blauen Linien nur sehr schwach waren. Wir konnten die blauen nur ohne das Fadenkreuz erkennen, weshalb das einstellen ungenauer war als bei den anderen. Aber alle Messungen befinden sich im Rahmen von Messungenauigkeiten. 


# Aufgabe 3
In Aufgabe 3 soll das Auflösungsvermögen des Gitters berechnet werden. Die Formel dafür kann aus der Experimentierbeschreibung entnommen werden.
$$A = \frac{\lambda}{\Delta \lambda}=kN$$
Wobei $k$ die Beugungsordnung und $N$ die Anzahl der beleuchteten Spalten ist. In der Aufgabestellung steht expilzit, dass, die wirksame Gitterbreite $b=50\text{mm}$ ($\pm 0.5%$) beträgt. Daraus lassen sich die Anzahl der Spalten ermitteln, indem man die wirksame Gitterbreite durch die Gitterkonstante mit Korrekturwinkel $g = 1.544 \cdot 10^{-6}$ teilt.
So ergibt sich die Formel der Auflösungsvermögen, welche wir nutzen mit, $A = k \frac{b}{g}$.


| Beugungsordnung $k$ | Auflösungsvermögen $A$ |
|---------------------|------------------------|
| $k = 1$             | 32383.420 $(\pm 169)$  |
| $k = 2$             | 64766.839 $(\pm 338)$  |
| $k = 3$             | 97159.259 $(\pm 507)$  |

: Auflösungsvermögen für verschiedenen Beugungsordnungen $k = 1, 2, 3$

## Fehlerbetrachtung und Diskussion
Mit den relativen Fehlern, den Größen, Gitterkosntante $g$ und Gitterbreite $b$ ergibt sich auch für das Auflösungsvermögen ein relativer Forpflanzungsfehler.

$$\frac{\delta A}{A} = \frac{\delta N}{N} = \sqrt{\left( \frac{\delta g}{g} \right)^2\left( \frac{\delta b}{b} \right)^2} = \sqrt{0.005^2+0.0015^2} \approx 5.22 \cdot 10^{-3}$$

Dieser wurde benutzt um in der Tabelle den absoluten Fehler für den Wert anzugeben. Dieser ist klein gegenüber dem Auflösungsvermögen.